File: C++/level3/2xn타일링.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(int n) {
    int array[60000];

    array[0] = 1;
    array[1] = 2;

    for(int i=2; i<n; i++){
        array[i] = (array[i-1] + array[i-2])%1000000007;
    }

    return array[n-1];
}

File: JavaScript/level3/2xn타일링.js
function solution(n) {
    let first = 1;
    let second = 2;
    
    if(n === 1)
        return first;
    
    if(n === 2)
        return second;
    
    for(let i=2; i<n; i++){
        let temp = second%1000000007;
        second += first;
        first = temp;
    }
    
    return second%1000000007;
}